pkg/stanza/fileconsumer/file_other.go

```diff
@@ -34,20 +36,24 @@ OUTER:
 				continue
 			}
 
-			// At this point, we know that the file has been rotated. However, we do not know
-			// if it was moved or truncated. If truncated, then both handles point to the same
-			// file, in which case we should only read from it using the new reader. We can use
+			// At this point, we know that the file has been rotated out of the matching pattern.
+			// However, we do not know if it was moved or truncated.
+			// If truncated, then both handles point to the same file, in which case
+			// we should only read from it using the new reader. We can use
 			// the Validate method to ensure that the file has not been truncated.
 			if !oldReader.Validate() {
+				m.set.Logger.Debug("File has been rotated(truncated)", zap.String("path", newReader.GetFileName()))
```

Comment:
I suppose it doesn't technically matter but do you think it would be slightly more readable to use `oldReader` here, simply because we are communicating to the reader about the old file that is no longer here?